File: androidapp/app/src/main/java/com/example/lanweather/data/model/Current.kt
package com.example.lanweather.data.model

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
class Current (
    val temperature: Double?,
    val maxTemperature: Double?,
    val minTemperature: Double?,
    val relativeHumidity: Int?,
    val windChill: Double?,
    val windDirection: Double?,
    val windSpeed: Double,
    val windGust: Double?,
    val weather: Weather?,
    val probabilityOfPrecipitation: Int?
)

File: flutterapp/lib/models/current.dart
import 'package:flutterapp/models/weather.dart';
import 'package:json_annotation/json_annotation.dart';

part 'current.g.dart';

@JsonSerializable()
class Current {
  final double? temperature;
  final double? maxTemperature;
  final double? minTemperature;
  final int? relativeHumidity;
  final double? windChill;
  final double? windDirection;
  final double? windSpeed;
  final double? windGust;
  final Weather? weather;
  final double? probabilityOfPrecipitation;

  Current(
    this.temperature,
    this.maxTemperature,
    this.minTemperature,
    this.relativeHumidity,
    this.windChill,
    this.windDirection,
    this.windSpeed,
    this.windGust,
    this.weather,
    this.probabilityOfPrecipitation,
  );

  factory Current.fromJson(Map<String, dynamic> json) =>
      _$CurrentFromJson(json);

  Map<String, dynamic> toJson() => _$CurrentToJson(this);
}
